### PR TITLE
Chrono types for Timestamp

### LINKED ISSUE #35: Create Timestamp from chrono type
It would be nice to implement `From` and `To` for [Chrono](https://crates.io/crates/chrono) types and the `influxdb::Timestamp`. Bonus points if it's behind a `chrono` feature gate.

### PR DESCRIPTION
**Pull request by durangatan**
## Description
Fixes #35.
- Each branch of the `Into` implementation casts everything to a `DateTime<Utc>` with at least millisecond precision. I figure the caller can cast the value to a timezone if needed.
-  I'm not sure if I'm representing `Timestamp`s the preferred way in UTC-land, because I'm not super familiar with how influxdb works. For example, `Timestamp::Hours(3)` probably doesn't mean three hours from the epoch, but for now I kept it consistent.
- The `From<DateTime>` implementation creates a `Timestamp::Milliseconds` from a given date. It will work for any DateTime object that implements `TimeZone`.
- these implementations are behind the `chrono_timestamps` compilation feature.

I hope this is sort of what you're looking for; let me know if you want me to take it in a different direction or implement more chrono converters.

### Checklist
- [x] Formatted code using `cargo fmt --all`
- [x] Linted code using clippy `cargo clippy --all-targets --all-features -- -D warnings`
- [ ] Updated README.md using `cargo readme > README.md`
- [x] Reviewed the diff. Did you leave any print statements or unnecessary comments?
- [x] Any unfinished work that warrants a separate issue captured in an issue with a TODO code comment

### CODE REVIEW

File: src/query/mod.rs [hunk 1]
```diff
@@ -19,14 +19,21 @@
...
+
 pub mod read_query;
 pub mod write_query;
 
 use std::fmt;
 
 use crate::{Error, ReadQuery, WriteQuery};
 
-#[derive(PartialEq)]
+#[derive(PartialEq, Debug)]
```
Comment (durangatan) [inline src/query/mod.rs]: need the Debug trait for the tests to work properly

File: src/query/mod.rs [hunk 2]
```diff
@@ -48,6 +55,43 @@ impl fmt::Display for Timestamp {
     }
 }
 
+#[cfg(any(test, feature = "chrono_timestamps"))]
+impl Into<DateTime<Utc>> for Timestamp {
+    fn into(self) -> DateTime<Utc> {
+        match self {
+            Timestamp::Now => Utc::now(),
+            Timestamp::Hours(h) => {
+                let millis = h * 60 * 60 * 1000;
```
Comment (msrd0) [inline src/query/mod.rs]: In the tests you are using `MINUTES_PER_HOUR` and the like, wouldn't it make sense to do the same here? Or maybe we can even use `chrono::Duration` or similar for this?

File: src/query/mod.rs [hunk 3]
```diff
@@ -157,7 +201,14 @@ pub enum QueryType {
 
 #[cfg(test)]
 mod tests {
+    extern crate chrono;
```
Comment (msrd0) [inline src/query/mod.rs]: This should be feature-gated (see above)

File: src/query/mod.rs [hunk 4]
```diff
@@ -48,6 +55,43 @@ impl fmt::Display for Timestamp {
...
+    }
+}
+
+#[cfg(any(test, feature = "chrono_timestamps"))]
+impl<T> From<DateTime<T>> for Timestamp
+where
+    T: TimeZone,
+{
+    fn from(date_time: DateTime<T>) -> Self {
+        Timestamp::Milliseconds(date_time.timestamp_millis() as usize)
```
Comment (Empty2k12) [inline src/query/mod.rs]: Better use `Timestamp:Nanoseconds` here, according to my comment.

File: src/query/mod.rs (hunk 1 above)
Comment (msrd0) [reply src/query/mod.rs]: `Copy` requires `Clone`, even if cloning is as cheap as copying

Comment (Empty2k12) [reply src/query/mod.rs]: I meant both, but it does not hurt to have them :)

Comment (msrd0) [reply src/query/mod.rs]: Ok well then I don't get your comment @Empty2k12

**Comment by msrd0** [comment]
Btw, right now you only implement `DateTime<Utc>` and I guess this makes sense not to have arbitrary Timezones but have the user convert them first. But would it make sense to support `NaiveDateTime` if the user explicitly don't care about timezones?

File: src/query/mod.rs (hunk 4 above)
Comment (msrd0) [reply src/query/mod.rs]: Also, [`timestamp_millis`](https://docs.rs/chrono/0.4.9/chrono/offset/trait.TimeZone.html#method.timestamp_millis) can panic while `timestamp_nanos` can't.

**Comment by durangatan** [comment]
I chose UTC because it is a "true unix timestamp". 
from the docs for `NaiveDateTime.timestamp_nanos`: 
```Returns the number of non-leap nanoseconds since midnight on January 1, 1970.

Note that this does not account for the timezone! The true "UNIX timestamp" would count seconds since the midnight UTC on the epoch.```

lmk if I should change the implementation to NaiveDateTime

**Comment by msrd0** [comment]
I see your point, but for some user who doesn't care about the timezone and interprets the `Utc` time as if it was their local one, it might make sense to also support `NaiveDateTime`. @Empty2k12 what do you think?

**Comment by durangatan** [comment]
Based on the above comments in the thread, I think it's a safe assumption that a user will expect an influxDB timestamp to represent a UTC datetime, not a timezoned time.
"Time in InfluxDb is UTC and an Unix nanosecond timestamp by default. We can set other precisions (which is done using the Timestamp enum)."

**Comment by Empty2k12** [comment]
I would say that supporting UTC only is reasonable, as InfluxDb makes that assumption for us and we should strive to behave similar to it.

File: Cargo.toml [hunk 5]
```diff
@@ -20,7 +20,12 @@ tokio = "0.1.20"
 failure = "0.1.5"
 serde = { version = "1.0.92", optional = true }
 serde_json = { version = "1.0", optional = true }
+chrono = { version = "0.4.9", optional = true }
+
+[dev-dependencies]
```
Comment (Empty2k12) [inline Cargo.toml]:
```suggestion

```

File: Cargo.toml [hunk 6]
```diff
@@ -21,7 +21,12 @@ tokio = "0.1.20"
...
+chrono = { version = "0.4.9" }
```
[the remark this answers is not in the document]
Comment (Empty2k12) [reply Cargo.toml]:
```suggestion

```

File: src/query/mod.rs [hunk 7]
```diff
@@ -48,6 +60,47 @@ impl fmt::Display for Timestamp {
     }
 }
 
+#[cfg(any(test, feature = "chrono_timestamps"))]
```
Comment (Empty2k12) [inline src/query/mod.rs]:
```suggestion
#[cfg(feature = "chrono_timestamps")]
```

File: src/query/mod.rs [hunk 8]
```diff
@@ -48,6 +60,47 @@ impl fmt::Display for Timestamp {
...
+            Timestamp::Nanoseconds(nanos) => Utc.timestamp_nanos(nanos.try_into().unwrap()),
+            Timestamp::Microseconds(mis) => {
+                let nanos = mis / 10000;
+                Utc.timestamp_nanos(nanos.try_into().unwrap())
+            }
+        }
+    }
+}
+
+#[cfg(any(test, feature = "chrono_timestamps"))]
```
Comment (Empty2k12) [inline src/query/mod.rs]:
```suggestion
#[cfg(feature = "chrono_timestamps")]
```

File: Cargo.toml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Empty2k12) [reply Cargo.toml]: @durangatan this one and we're ready to merge ;)

**Pull request merged**